src/ripple/consensus/Validations.h

```diff
@@ -207,20 +192,32 @@ class Validations
         decay_result_t<decltype (&Validation::ledgerID)(Validation)>;
     using NodeKey = decay_result_t<decltype (&Validation::key)(Validation)>;
     using NodeID = decay_result_t<decltype (&Validation::nodeID)(Validation)>;
-    using ValidationMap = hash_map<NodeKey, Validation>;
+    
 
     using ScopedLock = std::lock_guard<MutexType>;
 
     // Manages concurrent access to current_ and byLedger_
     MutexType mutex_;
 
+    //! For the most recent validation, we also want to store the ID
+    //! of the ledger it replaces
+    struct ValidationAndPrevID
+    {
+        ValidationAndPrevID(Validation const& v) : val{v}, prevLedgerID{0}
```

Comment:
I agree that is still worth addressing, but would prefer to do as an independent change.